addons/product/migrations/12.0.1.2/pre-migration.py

```diff
@@ -24,13 +27,27 @@
 _table_renames = [
     ('product_attribute_line', 'product_template_attribute_line'),
     ('product_attribute_price', 'product_template_attribute_value'),
+    ('product_attribute_line_product_attribute_value_rel',
+     'product_attribute_value_product_template_attribute_line_rel'),
 ]
 
 
+def avoid_new_constraint_in_product_template_attribute_line(cr):
+    openupgrade.logged_query(
```

Comment:
Why this?

Reply:
In v11 exists this contrain:
```python
        if any(line.value_ids > line.attribute_id.value_ids for line in self):
            raise ValidationError(_('Error ! You cannot use this attribute with the following value.'))
```
In v12, it has changed to:
```python
        if any(not line.value_ids or line.value_ids > line.attribute_id.value_ids for line in self):
            raise ValidationError(_('You cannot use this attribute with the following value.'))
```

It means that you only can create template attribute lines if and only if they are linked to an attribute value or more.

Reply:
Yeah, but you are destructing information with this, so we shouldn't unless it blocks something. In this case, I see it's a Python constraint, so it's not too much problem to let it.

Reply:
You are not destructing "information" in this case, it's just cleaning unused and useless data. Lines without an attribute value are pointless, that's why odoo added that new constrain.

Reply:
OK, got it